splunk_connect_for_snmp/gopoller/snmp.go

```diff
@@ -73,44 +52,21 @@ func ToString(value interface{}, vtype gosnmp.Asn1BER) string {
 	}
 }
 
-func getVersion(version string) gosnmp.SnmpVersion {
-	switch version {
-	case "1":
-		return gosnmp.Version1
-	case "2c":
-		return gosnmp.Version2c
-	default:
-		return gosnmp.Version3
-	}
-}
-
-func PerformGet(authData string, target string, community string, oids OidSlice, port int, ignoreNonIncreasingOid bool, version string) ([]Pdu, error) {
-	var appOpts = map[string]interface{}{"c": !ignoreNonIncreasingOid}
-	params := &gosnmp.GoSNMP{
-		Target:             target,
-		Port:               uint16(port),
-		Transport:          "udp",
-		ContextEngineID:    "", //defined in contextData, in Go only for v3
-		ContextName:        "", //defined in contextData, in Go only for v3
-		MaxRepetitions:     1,
-		NonRepeaters:       10,
-		AppOpts:            appOpts, //set AppOpts to c if ignoreNonIncreasingOid=false, no c if ignoreNonIncreasingOid=true
-		Community:          community,
-		Version:            getVersion(version),               // have to set to correct version and add necessary params for v3
-		Timeout:            time.Duration(1800) * time.Second, // timeout of one request/response, possibly not the updconnectiontimeout??
-		ExponentialTimeout: true,
-		//Logger:             gosnmp.NewLogger(log.New(os.Stdout, "", 0)), //for now logging to stdout for debuging
+func PerformGet(authData SnmpV3StringAuthData, target string, community string, oids OidSlice, port int, ignoreNonIncreasingOid bool, version string) ([]Pdu, error) {
+	params, err := getGoSnmp(target, port, ignoreNonIncreasingOid, version, community, authData, 20)
+	if err != nil {
+		return nil, fmt.Errorf("Error while creating gosnmp.GoSNMP:  %v", err)
 	}
 
-	err := params.Connect()
+	err = params.Connect()
 	if err != nil {
 		return nil, fmt.Errorf("error connecting to %s: %v", target, err)
 	}
 	defer params.Conn.Close()
 
 	res, err := params.Get(oids)
 	if err != nil {
-		return nil, fmt.Errorf("walk Error: %v", err)
+		return nil, fmt.Errorf("Get Error: %v", err)
```

Comment:
Maybe this error message could be better, or just return the error. It could be also changed for bulk.